tests/events/EventCard/menuDropdown.test.tsx

```diff
@@ -0,0 +1,111 @@
+import { TestEventItems } from "@lib/events/Event"
+import { fireEvent, render, screen } from "@testing-library/react-native"
+import EventItem from "@components/eventCard/EventItem"
+import React from "react"
+import { captureAlerts } from "../../helpers/Alerts"
+import { MenuProvider } from "react-native-popup-menu"
+import MenuDropdown from "@components/eventCard/MenuDropdown"
+
+jest.mock("react-native-popup-menu", () => ({
+  Menu: "Menu",
+  MenuProvider: "MenuProvider",
+  MenuOptions: "MenuOptions",
+  MenuOption: "MenuOption",
+  MenuTrigger: "MenuTrigger"
+}))
+
+describe("eventUI Component Tests", () => {
+  beforeEach(() => {
+    jest.resetAllMocks()
+  })
+
+  it("Event menu displays delete option when user is the host", () => {
+    render(
+      <MenuProvider>
+        <MenuDropdown isEventHost={true} />
+      </MenuProvider>
+    )
+
+    fireEvent.press(screen.getByTestId(moreButtonLabel))
+    expect(screen.queryByText("Delete")).toBeDefined()
+    expect(screen.queryByText("Report")).toBeNull()
+  })
+
+  it("Event menu displays report option when user isn't the host", () => {
+    render(
+      <MenuProvider>
+        <MenuDropdown isEventHost={false} />
+      </MenuProvider>
+    )
+
+    fireEvent.press(screen.getByTestId(moreButtonLabel))
+    expect(screen.queryByText("Report")).toBeDefined()
+    expect(screen.queryByText("Delete")).toBeNull()
+  })
+
+  it("Clicking delete opens alert", () => {
+    render(
+      <MenuProvider>
+        <MenuDropdown isEventHost={true} />
+      </MenuProvider>
+    )
+    fireEvent.press(screen.getByTestId(moreButtonLabel))
+    fireEvent.press(screen.getByText("Delete"))
+
+    expect(alertPresentationSpy).toHaveBeenCalled()
+  })
+
+  //* * Right now event isn't removed */
+  it("Clicking delete on alert closes alert and removes event", () => {
+    render(
+      <MenuProvider>
+        <MenuDropdown isEventHost={true} />
+      </MenuProvider>
+    )
+    fireEvent.press(screen.getByTestId(moreButtonLabel))
+    fireEvent.press(screen.getByText("Delete"))
+
+    dismissConfirmDelete()
+    expect(screen.queryByText("Delete this event?")).toBeNull()
```

Comment:
One suggestion, since you've extracted `dismissConfirmDelete` which taps the delete button, you may also want to do this for the `fireEvent` stuff as well. In other words, I think the test becomes more self-explanatory when it looks like this (using a hypothetical `Events` interface):

```ts
render(...)
attemptDeletion()
await confirmDeletion()
// Assert that the event was actually deleted in some way
```

rather than

```ts
render(...)
fireEvent.press(screen.getByTestId(moreButtonLabel))
fireEvent.press(screen.getByText("Delete"))
dismissConfirmDelete()
// Assert that the event was actually deleted in some way
```

The former just presents the test as more of a set of requirements rather than interacting directly with implementation details.

Reply:
Thanks for the suggestion! It does look a lot cleaner and more intuitive to do it that way.